src/radical/entk/execman/mock/resource_manager.py

```diff
@@ -29,7 +29,7 @@ class ResourceManager(Base_ResourceManager):
 
     # --------------------------------------------------------------------------
     #
-    def __init__(self, resource_desc, sid, rts_config={}):
+    def __init__(self, resource_desc, sid, rts_config):
 
```

Comment:
Is the intent of that change to get in sync with the other RM implementations?

Reply:
RM? I am not sure what that is. I did the change because of this message from pylint:
```
 W0102: Dangerous default value {} as argument (dangerous-default-value)
```

Reply:
`RM` - the ResourceManager whose constuctor signature you changed.

The fix for that linting error is:
```py
def __init__(self, resource_desc, sid, rts_config=None):
    if rts_config is None:
        rts_config = dict()
```

The reason for that is some more python by-value / by-reference / data localization stupidity (really, it's braindead).  The change you did changes the signature, which you may want to avoid.  The above does not change the signature, but delays the problematic default value assignment.

Reply:
It's already happening in the base class: https://github.com/radical-cybertools/radical.entk/blob/fix/file_linting/src/radical/entk/execman/base/resource_manager.py#L34

Reply:
I really don't know what the signature *should* be - just saying that the syntactic-neutral change for the linting error would be a bit different.  Anyway, happy to leave that choice to you.